word_search.py

```diff
@@ -0,0 +1,102 @@
+'''
+Created on 9 June 2017
+
+@author: Eva
+'''
+from queue import Queue
+
+class WordSearch:
+    
+    def __init__(self):
+        return
+        
+    def words_found_in_grid(self, rowNo, columnNo, grid, dictionary):
+        """Uses bfs to find valid words from a grid according to a given dictionary 
+        
+            Args:
+                rowNo: number of rows in the grid
+                columnNo: number of columns in the grid
+                grid: two-dimensional array of characters
+                dictionary: a dictionary object that holds the valid words and prefices
+            
+            Returns:
+                A list of words that were found on the grid
+        """
+        self.__pathsTaken = []
+        self.__foundWords = []
+        self.__words_to_search = Queue()
+        row = 0
+        column = 0
+        
+        self.__words_to_search.put([grid[column][row], column, row, []])
+        
+        while not self.__words_to_search.empty():
+            currentArray = self.__words_to_search.get()
+            currentWord = currentArray[0]
+            column = currentArray[1]
+            row = currentArray[2]
+            positions = currentArray[3]
+            
+            if dictionary.isPrefix(currentWord):
+                tempPositions = positions.copy()
+                tempPositions.append([column, row])
+                if tempPositions in self.__pathsTaken: continue
+                
+                if dictionary.isWord(currentWord) and currentWord not in self.__foundWords:
+                    self.__foundWords.append(currentWord)
+                    newPositions = positions.copy()
+                    newPositions.append([column, row])
+                    self.__pathsTaken.append(newPositions)
+            
+                if row+1 < rowNo and [column, row+1] not in positions:
```

Comment:
This is a long list of similar, but slightly different `if`-statements.
Can we find a way to avoid copy-and-pasting these each time?
Can we make this code parametric, so that we can generate all the combination we need instead of listing explicitly?